### PR TITLE
Add async support for organizations

### PR DESCRIPTION
**Pull request by mattgd**
## Description
Add async support for organizations. This is the last module I felt like async support could absolutely be useful, in tandem with other modules.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: workos/types/webhooks/webhook.py [hunk 1]
```diff
@@ -1,11 +1,9 @@
-from typing import Generic, Literal, Union
```
Comment (mattgd) [inline workos/types/webhooks/webhook.py:1]: Cleaning up unused imports.

File: tests/test_organizations.py [hunk 2]
```diff
@@ -200,3 +202,142 @@ def test_list_organizations_auto_pagination_for_multiple_pages(
             list_function=self.organizations.list_organizations,
             expected_all_page_data=mock_organizations_multiple_data_pages["data"],
         )
+
+
+@pytest.mark.asyncio
+class TestAsyncOrganizations(OrganizationFixtures):
```
Comment (tribble) [inline tests/test_organizations.py:208]: This repetition is so grrrrr.

File: workos/organizations.py [hunk 3]
```diff
@@ -75,29 +78,39 @@ def create_organization(
         name: str,
         domain_data: Optional[Sequence[DomainDataInput]] = None,
         idempotency_key: Optional[str] = None,
-    ) -> Organization: ...
+    ) -> SyncOrAsync[Organization]:
+        """Create an organization
```
Comment (tribble) [inline workos/organizations.py:82]: Odd. I wonder why this is in the diff Is it actually missing?

**Review by tribble — APPROVED** [review]
Great Job

File: tests/test_organizations.py (hunk 2 above)
Comment (mattgd) [reply tests/test_organizations.py:208]: I know 😢 We need some more thought to strike the right balance between little repetition and and readability.

**Pull request merged**